### PR TITLE
staging and production

### PR DESCRIPTION
**Pull request by 1JW1**
## Describe this PR

Enabling the API to be pushed to staging and production in AWS

### *What changes have we introduced*

uncommented code in the config.yml to allow the api to be pushed to the staging and production accounts in AWS

#### _Checklist_

- [x] Code pipeline builds correctly

### *Follow up actions after merging PR*

Are there any next steps that need to addressed after merging this PR? Add them here.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -183,59 +183,59 @@ workflows:
...
-#             - terraform-init-and-apply-to-production
-#           filters:
-#             branches:
-#               only: master
+  check-and-deploy-staging-and-production:
+      jobs:
+      - build-and-test:
+          filters:
+            branches:
+              only: master
```
Comment (humulla) [inline .circleci/config.yml:191]: Please change to `release`

File: .circleci/config.yml [hunk 2]
```diff
@@ -183,59 +183,59 @@ workflows:
...
+      - assume-role-staging:
+          context: api-assume-role-staging-context
+          requires:
+              - build-and-test
+          filters:
+             branches:
+               only: master
```
Comment (humulla) [inline .circleci/config.yml:198]: Please change to `release`

File: .circleci/config.yml [hunk 3]
```diff
@@ -183,59 +183,59 @@ workflows:
...
+      - terraform-init-and-apply-to-staging:
+          requires:
+            - assume-role-staging
+          filters:
+            branches:
+              only: master
```
Comment (humulla) [inline .circleci/config.yml:204]: Please change to `release`

File: .circleci/config.yml [hunk 4]
```diff
@@ -183,59 +183,59 @@ workflows:
...
+      - deploy-to-staging:
+          requires:
+            - terraform-init-and-apply-to-staging
+          filters:
+            branches:
+              only: master
```
Comment (humulla) [inline .circleci/config.yml:210]: Please change to `release`

File: .circleci/config.yml [hunk 5]
```diff
@@ -183,59 +183,59 @@ workflows:
...
+          type: approval
+          requires:
+            - deploy-to-staging
+      - assume-role-production:
+          context: api-assume-role-production-context
+          requires:
+              - permit-production-terraform-release
+          filters:
+             branches:
+               only: master
```
Comment (humulla) [inline .circleci/config.yml:221]: Please change to `release`

File: .circleci/config.yml [hunk 6]
```diff
@@ -183,59 +183,59 @@ workflows:
...
+      - terraform-init-and-apply-to-production:
+          requires:
+            - assume-role-production
+          filters:
+            branches:
+              only: master
```
Comment (humulla) [inline .circleci/config.yml:227]: Please change to `release`

File: .circleci/config.yml [hunk 7]
```diff
@@ -183,59 +183,59 @@ workflows:
...
+      - permit-production-release:
+          type: approval
+          requires:
+            - deploy-to-staging
+          filters:
+            branches:
+              only: master
```
Comment (humulla) [inline .circleci/config.yml:234]: Please change to `release`

File: .circleci/config.yml [hunk 8]
```diff
@@ -183,59 +183,59 @@ workflows:
...
+      - deploy-to-production:
+          requires:
+            - permit-production-release
+            - terraform-init-and-apply-to-production
+          filters:
+            branches:
+              only: master
```
Comment (humulla) [inline .circleci/config.yml:241]: Please change to `release`

**Review by humulla — COMMENTED** [review]
Before you merge this branch please do the following:

- Get the latest from master locally
- Create a new branch called `release`
- Push the release branch to github

File: .circleci/config.yml (hunk 1 above)
Comment (1JW1) [reply .circleci/config.yml:191]: Hey @humulla , the latest changes were pulled. Ill make a new branch called release instead and create a new PR. As for the changes, should all branches for staging and prod be `only: release` or just the ones you've marked on this PR?

**Review by humulla — APPROVED** [review]

**Pull request merged**